From 1ea4541ceeed09a01d18b3bd75a09bd6d18bdb84 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Mon, 18 Nov 2019 12:43:53 +0100
Subject: [PATCH] build: Add flag to disable update checker

The update checking preferences are still present in the UI though.
---
 Makefile                          | 4 ++++
 include/StSocket/StCheckUpdates.h | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/Makefile b/Makefile
index d89b64ae..3acb47f8 100644
--- a/Makefile
+++ b/Makefile
@@ -159,6 +159,10 @@ else ifneq (,$(findstring mingw,$(CCMACHINE)))
 TARGET_ARCH2 = x86
 endif
 
+ifeq ($(DISABLE_UPDATER),)
+EXTRA_CXXFLAGS += -DWITH_UPDATER=1
+endif
+
 # to activate debug build
 #EXTRA_CXXFLAGS = -DST_DEBUG_LOG_TO_FILE=\"/sdcard/Android/data/com.sview/files/sview.log\" -DST_DEBUG
 #EXTRA_CXXFLAGS += -DST_DEBUG_GL
diff --git a/include/StSocket/StCheckUpdates.h b/include/StSocket/StCheckUpdates.h
index 9c111019..b133bba4 100644
--- a/include/StSocket/StCheckUpdates.h
+++ b/include/StSocket/StCheckUpdates.h
@@ -86,6 +86,7 @@ class StCheckUpdates {
     }
 
     void init() {
+#ifdef WITH_UPDATER
         /// TODO (Kirill Gavrilov#1) instability in browsers???
         myMutex.lock();
         if(!myThread.isNull()) {
@@ -94,8 +95,10 @@ class StCheckUpdates {
         }
         myThread = new StThread(checkUpdatesThread, this, "StCheckUpdates");
         myMutex.unlock();
+#endif
     }
 
+#ifdef WITH_UPDATER
     static bool checkUpdates() {
         const char     THE_URL_TEMPLATE[] = "http://www.sview.ru/updates/?appl=sView&year=%d&month=%d&rStatus=%d&rSubVer=%d";
         const StString THE_TO_UPDATE      = "<update>yes</update>";
@@ -125,6 +128,7 @@ class StCheckUpdates {
         aThis->setInitialized();
         return SV_THREAD_RETURN 0;
     }
+#endif
 
         private:
 
-- 
2.23.0

